src/goto-instrument/generate_function_bodies.cpp

```diff
@@ -74,15 +74,13 @@ class assume_false_generate_function_bodiest : public generate_function_bodiest
   {
     auto const &function_symbol = symbol_table.lookup_ref(function_name);
     // NOLINTNEXTLINE
-    auto add_instruction = [&]() {
-      auto instruction = function.body.add_instruction();
+    auto add_instruction = [&](goto_programt::instructiont &&i) {
+      auto instruction = function.body.add(std::move(i));
       instruction->source_location = function_symbol.location;
       return instruction;
     };
```

Comment:
Does this lambda still have any value left in it? The `make_*` calls would now also take a source location.